gulpfile.babel.js

```diff
@@ -128,12 +143,87 @@ const buildConfig = (invocationArgs, sourceRoot, testRoot, buildRoot) => {
 
 const config = buildConfig(process.argv, 'source', 'test', 'build');
 
+const httpCache = new Keyv({
+  store: new KeyvFile({
+    filename: `${config.dir.build.cache}/http.json`,
+  }),
+});
+
 // Shared config tasks
 
 const cleanSharedConfig = () => del(config.files.src.derivedConfigs);
 
 export const distributeSharedConfig = gulp.series(cleanSharedConfig, distributeConfig);
 
+// Font tasks
+
+const cleanFonts = () => del(config.files.src.fonts.concat([config.dir.src.sassFonts]));
+
+const compileFonts = () => {
+  fs.ensureDirSync(config.dir.src.fonts);
+
+  const fonts = yaml.safeLoad(fs.readFileSync(config.files.src.fontsDefinition));
+
+  const files = fonts.reduce(
+    (carry, font) => {
+      font.files.forEach(file => {
+        const uri = url.resolve(font.base, file.path);
+        if (!(uri in carry)) {
+          carry[uri] = [];
+        }
+        carry[uri].push({
+          fontDisplay: 'fallback',
+          fontFamily: font.name,
+          fontStyle: file.style || 'normal',
+          fontWeight: file.weight || 400,
+          subsetMap: font.subsets,
+        });
+      });
+
+      return carry;
+    }, {},
+  );
+
+  const fontCache = `${config.dir.build.cache}/fonts`;
+  fs.removeSync(fontCache);
+
+  return Promise.all(Object.keys(files).map(uri =>
+    download(uri, {cache: httpCache})
+      .then(data => tempWrite(data, path.basename(uri)))
+      .then(fontFile => {
+        return Promise.all(files[uri].map(file => {
+          file.fontFile = fontFile;
+          file.outputFolder = fontCache;
+          return fontRanger(file);
+        }))
+          .finally(() => fs.promises.unlink(fontFile));
+      })))
+    .then(() => copy(fontCache, config.dir.src.sassFonts, {
+        filter: '*.css',
+        rename: basename => path.basename(basename, '.css') + `.scss`,
+        transform: () => replaceStream('url(\'', 'url(\'#{$fonts-path}/'),
+      },
+    ))
+    .then(() => copy(fontCache, config.dir.src.fonts, {filter: '*.woff2'}));
+};
+
+const compileFontLicenses = () => {
+  fs.ensureDirSync(config.dir.src.fonts);
+
+  const fonts = yaml.safeLoad(fs.readFileSync(config.files.src.fontsDefinition));
+
+  const licenses = fonts.reduce((licenses, font) => licenses.add(url.resolve(font.base, font.license)), new Set());
+
+  return Promise.all([...licenses].map(uri =>
+    download(uri, {cache: httpCache})
+      .then(data => data.toString()),
+  ))
+    .then(licenses => [...new Set(licenses)])
+    .then(licenses => fs.promises.writeFile(`${config.dir.src.fonts}/LICENSE`, licenses.join('\n\n\n\n\n---\n\n\n\n\n\n')));
```

Comment:
If there's more than one license, is it clear to which font(s) a license applies? Do we need something in the combined `LICENSE` describing which font(s) a particular license applies to?

Reply:
Relates to https://github.com/libero/pattern-library/pull/86#discussion_r280039007. Not sure what you should do if there's multiple licenses... /cc @BlueReZZ

Reply:
Convention is to either combine LICENSE files by concatenation in the one in the root, or have one per directory and store differently licensed content in different directories

Reply:
This combines them into one file with the divider. No obvious way to say what fonts are under which license, but 🤷‍♂ (won't happen until we use some other font under a different license too, so unlikely).

Reply:
Happy that this is okay for now.